skbuild/cmaker.py

```diff
@@ -151,27 +171,61 @@ def get_python_version():
 
         return python_version
 
-    @staticmethod
+    # NOTE(opadron): The try-excepts raise the cyclomatic complexity, but we
+    # need them for this function.
+    @staticmethod  # noqa: C901: Complexity
     def get_python_include_dir(python_version):
         # determine python include dir
         python_include_dir = sysconfig.get_config_var('INCLUDEPY')
 
         # if Python.h not found (or python_include_dir is None), try to find a
         # suitable include dir
-        found_python_h = os.path.exists(
-            os.path.join(python_include_dir, 'Python.h'))
-
-        if python_include_dir is None or not found_python_h:
+        found_python_h = (
+            python_include_dir is not None or
+            os.path.exists(os.path.join(python_include_dir, 'Python.h'))
+        )
 
-            candidate_prefixes = [
-                os.path.dirname(sysconfig.get_config_var('INCLUDEPY')),
-                sysconfig.get_config_var('INCLUDEDIR'),
-                os.path.dirname(sysconfig.get_path('include')),
-                os.path.dirname(sysconfig.get_path('platinclude')),
-                os.path.join(sysconfig.get_python_inc(),
-                             ".".join(map(str, sys.version_info[:2]))),
-                sysconfig.get_python_inc()
-                ]
+        if not found_python_h:
+            candidate_prefixes = []
+
+            # NOTE(opadron): these possible prefixes must be guarded against
+            # AttributeErrors and KeyErrors because they each can throw on
+            # different platforms or even different builds on the same platform.
```

Comment:
Since the documentation indicate `None` is returned, could you revert these changes ?

* https://docs.python.org/2/library/sysconfig.html#sysconfig.get_config_var
* https://docs.python.org/3/library/sysconfig.html#sysconfig.get_config_var

On Linux, no error:
```
$ python -c "import sysconfig; print(sysconfig.get_config_var('PYTHONPATH'))"
:plat-x86_64-linux-gnu:lib-tk:lib-old

$ python -c "import sysconfig; print(sysconfig.get_config_var('Invalid'))"
None
```

On MacOSx, no error:
```
$ python -c "import sysconfig; print(sysconfig.get_config_var('Invalid'))"
None
$ python -c "import sysconfig; print(sysconfig.get_config_var('PYTHONPATH'))"
:plat-darwin:plat-mac:plat-mac/lib-scriptpackages:../../Extras/lib/python:lib-tk:lib-old
```

On windows, I expect the same.